scipy/stats/tests/test_multivariate.py

```diff
@@ -546,6 +671,25 @@ def test_cdf_signs(self):
         cdf = multivariate_normal.cdf(b, mean, cov, lower_limit=a)
         assert_allclose(cdf, cdf[0]*expected_signs)
 
+    def test_mean_cov(self):
+        # test the interaction between a Covariance object and mean
+        P = np.diag(1 / np.array([1, 2, 3]))
+        cov_object = _covariance.CovViaPrecision(P)
+
+        message = "`cov` represents a covariance matrix in 3 dimensions..."
+        with pytest.raises(ValueError, match=message):
+            multivariate_normal.entropy([0, 0], cov_object)
+
+        with pytest.raises(ValueError, match=message):
+            multivariate_normal([0, 0], cov_object).cdf([1, 2, 3])
```

Comment:
```suggestion
            multivariate_normal([0, 0], cov_object)
```

I think initializing the object with dimension mismatch should raise the error, not calling `cdf`.